DEVELOPING.md

```diff
@@ -0,0 +1,138 @@
+1. Set the value of `Webhook URL` to `http://${APP_HOSTNAME}/`
+1. Set Repository metadata, Repository administration, Repository content and Organization members to read only (this is only used for your org so it's safe)
+1. Save and download the private key. It will result in a file like `app-name.date.private-key.pem`
+1. Copy the private key to `${HABITAT_SRC_ROOT}/.secrets/builder-github-app.pem`
+1. Record the the client-id, client-secret, app_id and public page link (in the left sidebar). These will be used for the `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`, `GITHUP_APP_ID` and `GITHUB_PUBLIC_LINK` build variables (seen below).
+
+### Create app env file
+
+1. Create `${HABITAT_SRC_ROOT}/.secrets/habitat-env`
+1. Add
+```
+  APP_HOSTNAME=localhost
+  GITHUB_API_URL=https://api.github.com
+  GITHUB_WEB_URL=https://github.com
+  GITHUB_CLIENT_ID=${GITHUB_CLIENT_ID}
+  GITHUB_CLIENT_SECRET=${GITHUB_CLIENT_SECRET}
+  GITHUB_APP_ID=${GITHUP_APP_ID}
+  GITHUB_APP_NAME="${GITHUB_PUBLIC_LINK}"
+```
+1. Save and close
+
+### Studio requirements
+
+1. Ensure you have run `hab setup` in your environment that will be executing the studio
+
+### Starting the services
+
+From either your VM or Mac:
+
+* `cd <source path>`
+* `direnv allow`
+* `hab studio enter`
+* `start-builder`
+
+### UI Setup
+
+If you are not doing UI development you just need to navigate to `${APP_HOSTNAME}/#/sign-in`
+
+If you are developing the UI:
+
+* Follow the instructions in the [Web UI README](https://github.com/habitat-sh/habitat/blob/master/components/builder-web/README.md) to get the Web UI running locally.
+* Open up a browser and navigate to http://localhost:3000/#/pkgs - you should see the Habitat UI running locally.
+* In the studio, you will need to run
+* * `ui-dev-mode` to swap out the github application for development on `localhost:3000`
+* * `upload_github_keys` to update the private key from your app with the new shared key for the app connected to `localhost:3000`
+* * Note: Make sure you have copied the private key as described [here](#GitHub OAuth Application)
+
+## Helper functions
+
+* `start-builder` - Run the builder cluster
+* `origin <name>` - Create the core origin. Default: core
+* `project` - Create a project (you can also configure this in the web UI)
+* `build-builder` - Build all the builder components
+* `build-<component>` - Ex: `build-router` will build the router component for development and load it
+* `dev_docs` - Print the full set of command docs
+
+### Create a project using the web UI
+
+1. Go the web UI that you used in the last step
+2. Go to the origins page, and select your origin
+3. Click on the 'Connect a plan file' button
+4. Click on 'Install Github App' button to install the Builder Dev app on your github account
+5. Go back to the Packages page (from Step 3), and follow the instructions to link the plan you want to build
```

Comment:
Note: your GH app must have access to the repo containing the plan file. Forking `habitat-sh/core-plans` is an easy way to test.

Reply:
👍